scripts/generate_schemas.py

```diff
@@ -200,32 +204,63 @@ def __init__(self, version, schemaDir, outputFile):
         self.schemaDir = schemaDir
         self.outputFile = outputFile
         self.classes = []
-        for f in glob.glob(os.path.join(self.schemaDir, "*.avsc")):
-            self.classes.append(SchemaClass(f))
+        for avscFile in glob.glob(os.path.join(self.schemaDir, "*.avsc")):
+            self.classes.append(SchemaClass(avscFile))
+        self.requestClassNames = \
+            [clazz.name for clazz in self.classes
+                if re.search('Search.*Request', clazz.name)]
+        self.responseClassNames = \
+            [clazz.name for clazz in self.classes
+                if re.search('Search.*Response', clazz.name)]
+        self.postSignatures = []
+        for request, response in zip(
+                self.requestClassNames, self.responseClassNames):
+            objname = re.search('Search(.*)Request', request).groups()[0]
+            url = '/{0}/search'.format(objname.lower())
+            tup = (url, request, response)
+            self.postSignatures.append(tup)
 
-    def writeHeader(self, f):
+    def writeHeader(self, outputFile):
         """
         Writes the header information to the output file.
         """
-        print('"""{0}"""'.format(HEADER_COMMENT), file=f)
-        print("from protocol import ProtocolElement", file=f)
-        print("import avro.schema", file=f)
-        print(file=f)
+        print('"""{0}"""'.format(HEADER_COMMENT), file=outputFile)
+        print("from protocol import ProtocolElement", file=outputFile)
+        print("import avro.schema", file=outputFile)
+        print(file=outputFile)
         versionStr = self.version[1:]  # Strip off leading 'v'
-        print("version = '{0}'".format(versionStr), file=f)
+        print("version = '{0}'".format(versionStr), file=outputFile)
 
     def write(self):
         """
         Writes the generated schema classes to the output file.
         """
-        with open(self.outputFile, "w") as f:
-            self.writeHeader(f)
+        with open(self.outputFile, "w") as outputFile:
+            self.writeHeader(outputFile)
             # Get the classnames and sort them to get consistent ordering.
-            names = [c.name for c in self.classes]
-            classes = dict([(c.name, c) for c in self.classes])
+            names = [clazz.name for clazz in self.classes]
```

Comment:
Ditto to preferring `cls` here - if you think it's important though, we can easily change the other places we're following this convention to `clazz` and put a note in the upcoming style guide.